src/Concordium/Client/GRPC2.hs

```diff
@@ -2168,19 +2167,19 @@ protoToTokenEvent event = do
             details <- (fromProto . CBorAsTokenEventDetails) (e ^. PLTFields.details)
             return $ TokenModuleEvent tokenId type' details
         ProtoPLT.TokenEvent'MintEvent e -> do
-            target <- Cbor.holderAccountAddress <$> fromProto (e ^. ProtoFieldsPLT.target)
+            target <- fromProto $ e ^. ProtoFieldsPLT.target
             amount <- fromProto $ e ^. ProtoFieldsPLT.amount
             return $ TokenMint tokenId target amount
         ProtoPLT.TokenEvent'BurnEvent e -> do
-            target <- Cbor.holderAccountAddress <$> fromProto (e ^. ProtoFieldsPLT.target)
+            target <- fromProto $ e ^. ProtoFieldsPLT.target
             amount <- fromProto $ e ^. ProtoFieldsPLT.amount
-            return $ TokenMint tokenId target amount
+            return $ TokenBurn tokenId target amount
         ProtoPLT.TokenEvent'TransferEvent e -> do
-            fromAccount <- Cbor.holderAccountAddress <$> fromProto (e ^. ProtoFieldsPLT.from)
-            toAccount <- Cbor.holderAccountAddress <$> fromProto (e ^. ProtoFieldsPLT.to)
+            fromAddress <- fromProto $ e ^. ProtoFieldsPLT.from
+            toAddress <- fromProto $ e ^. ProtoFieldsPLT.to
             amount <- fromProto $ e ^. ProtoFieldsPLT.amount
             memo <- fromProtoMaybe $ e ^. ProtoFields.maybe'memo
-            return $ TokenTransfer tokenId fromAccount toAccount amount memo
+            return $ TokenTransfer tokenId fromAddress toAddress amount memo
```

Comment:
I think it's a good idea to use record syntax, especially when constructors have many parameters and where there are multiple parameters of the same type (e.g. the from and to addresses). This makes it easier to catch any issues that might come from any changes.
```suggestion
            return TokenTransfer {
                    ettTokenId = tokenId,
                    ettFrom = fromAddress,
                    ettTo = toAddress,
                    ettAmount = amount,
                    ettMemo = memo
                }
```
(Typically, I would name the variables the same as the fields and use record wildcards with `TokenTransfer {..}`.)